### ISSUE TITLE
Images in markdown do not get pulled in

### ISSUE THREAD

**Issue by tsmcgrath** [vb64/markdown-pdf#25]
I am using markdown-pdf to pull in several existing markdown files with embedded images and write them to a single pdf. The separate markdowns display the images (with either a relative or absolute path) correctly. But, when I read them into the library with pdf.add_section the markdown comes in fine and converts to a pdf file but the image is not included.
Code:
`from markdown_pdf import MarkdownPdf
from markdown_pdf import Section

# create pdf
pdf = MarkdownPdf(toc_level=2)
# add section
pdf.add_section(Section("# Catchment ID 44193\n"))
# add 2nd section from markdown file
md = open('./markdown/Intro.md', 'r', newline='', encoding='utf-8-sig').read()
pdf.add_section(Section(md))

# set pdf properties
pdf.meta["title"] = "LOCA Report"
pdf.meta["author"] = "Tyson Broad"
# save pdf
pdf.save("./src/python/md2pdf_test3.pdf")
`
Image of the local markdown displaying image correctly:
![image](https://github.com/vb64/markdown-pdf/assets/1571883/f73b152a-bbdd-4cfb-9090-075d0bda7a29)

Problematic markdown text attached. 
[Intro.md](https://github.com/user-attachments/files/16120137/Intro.md)
Output PDF attached.
[md2pdf_test3.pdf](https://github.com/user-attachments/files/16120138/md2pdf_test3.pdf)

**Comment by tsmcgrath** [comment]
That doesn't seem to work. I also moved the markdown and image file local to the python script and that doesn't seem to work either. The markdown comes in ok, no image.

**Comment by tsmcgrath** [comment]
I can pull an image into it's on section on it's own. Without being inside markdown.

**Comment by tsmcgrath** [comment]
Sorry to be a problem child. This is a very useful library and I really appreciate your work.

**Comment by vb64** [comment]
Sorry, I really didn’t notice that in Intro.md you use `html` for the image, not `markdown`.

markdown-pdf is designed to handle `markdown`, not mixed markdown/html code.

Try changing the code for the image in Intro.md to the following:

```markdown
[Steep Rocky- Oak/Juniper/Woodland in foreground. Loamy Bottomland-San Saba River in background](titletest1.jpeg)
```

And place the file `titletest1.jpeg` in the current directory where your script is running from.

**Comment by tsmcgrath** [comment]
Tried that. Still no luck.
[md2pdf_test3.pdf](https://github.com/user-attachments/files/16120704/md2pdf_test3.pdf)

**Comment by vb64** [comment]
You can try the [example from the readme](https://github.com/vb64/markdown-pdf?tab=readme-ov-file#example) yourself. The generated pdf will contain an embedded image with the Python logo.

You need to clone the markdown-pdf repository to your host, open a console window, go to the root directory of the project (where the README.md file is located) and run

```bash
python makepdf.py README.md markdown_pdf.pdf
```

A markdown_pdf.pdf file will be created in this directory.

If successful, this will be a good starting point for writing the code you need.

**Comment by tsmcgrath** [comment]
I have successfully gotten the example to work without a problem. That's how I started. Inserting an image into the pdf using the library works fine. But if the image is referenced in the markdown file, even using a markdown reference and not html, it does not make it into the pdf. Even if the image correctly displays when previewing the markdown in a browser.

The markdown:
```
## The Upper San Saba River
![Steep Rocky- Oak/Juniper/Woodland in foreground. Loamy Bottomland-San Saba River in background](images/titletest1.jpeg)

The spring-fed San Saba River is a valuable resource to West-Central Texas. The relatively constant flows of the river
```

Image rendering correctly when referenced in markdown:
![image](https://github.com/vb64/markdown-pdf/assets/1571883/4fabe248-d1ad-431f-b7ad-23b3ba6120b9)

Resulting pdf:
[md2pdf_test3.pdf](https://github.com/user-attachments/files/16132174/md2pdf_test3.pdf)

**Comment by tsmcgrath** [comment]
If using a referenced image is just not included in the current functionality, fine. I accept that. I just wanted to make sure it was not a bug.

**Comment by seangrogan** [comment]
Hi, I know this is an older issue but I was having the same problem and I found a solution: 

I needed to add the full path to the image file in the `root=` argument, e.g. `c:/Users/tim/DocumentsLocal/Github/catch23/images/`

So my code in my specific project is 

```
def create_pdf_from_md(md_filepath, pdf_filepath):
    with open(md_filepath, 'r') as f:
        md_content = f.read()
   
    root = os.path.dirname(os.path.abspath(md_filepath))
    pdf = MarkdownPdf()
    pdf.add_section(Section(md_content, root=root))
    pdf.save(pdf_filepath)
```

**Comment by vb64** [comment]
Yes, the **root** argument is intended for cases where the embedded image files are not located in the current directory.
